apache-rat-core/src/test/java/org/apache/rat/config/exclusion/fileProcessors/GitIgnoreBuilderTest.java

```diff
@@ -125,13 +135,50 @@ public void test_RAT_335() {
         DocumentName candidate = DocumentName.builder()
                 .setName("/home/claude/apache/creadur-rat/apache-rat-core/target/test-classes/GitIgnoreBuilderTest/src/dir1/file1.log")
                 .setBaseName("home/claude/apache/creadur-rat/apache-rat-core/target/test-classes/GitIgnoreBuilderTest/src/").build();
-        System.out.println("Decomposition for "+candidate);
+        System.out.println("Decomposition for " + candidate);
 
-        assertThat(matcher.toString()).isEqualTo("matcherSet(or('included dir1/.gitignore', 'included .gitignore'), or('excluded dir1/.gitignore', **/.gitignore, 'excluded .gitignore'))");
+        assertThat(matcher.toString()).isEqualTo("matcherSet(or('included .gitignore', 'included .gitignore'), or('excluded .gitignore', **/.gitignore, 'excluded .gitignore'))");
 
         List<String> notMatching = Arrays.asList("README.txt", "dir1/dir1.md", "dir2/dir2.txt", "dir3/file3.log", "dir1/file1.log");
 
-        List<String> matching = Arrays.asList(".gitignore", "root.md", "dir1/.gitignore", "dir1/dir1.txt",  "dir2/dir2.md", "dir3/dir3.log");
+        List<String> matching = Arrays.asList(".gitignore", "root.md", "dir1/.gitignore", "dir1/dir1.txt", "dir2/dir2.md", "dir3/dir3.log");
+
+        assertCorrect(matcherSets, documentName.getBaseDocumentName(), matching, notMatching);
+    }
+
+    /**
+     * Test that exclusions from a global gitignore are also applied
+     *
+     * https://issues.apache.org/jira/browse/RAT-473
+     */
+    @Test
+    public void test_global_gitignore() {
+        URL globalGitIgnoreUrl = GitIgnoreBuilderTest.class.getClassLoader().getResource("GitIgnoreBuilderTest/global-gitignore");
+        String globalGitIgnore = globalGitIgnoreUrl.getFile();
+
+        GitIgnoreBuilder underTest = new GitIgnoreBuilder() {
+            @Override
+            protected String globalGitIgnore() {
+                return globalGitIgnore;
+            }
+        };
+        URL url = GitIgnoreBuilderTest.class.getClassLoader().getResource("GitIgnoreBuilderTest/src/");
+        File file = new File(url.getFile());
+
+        DocumentName documentName = DocumentName.builder(file).build();
+        List<MatcherSet> matcherSets = underTest.build(documentName);
+        DocumentNameMatcher matcher = MatcherSet.merge(matcherSets).createMatcher();
+
+        DocumentName candidate = DocumentName.builder()
+                .setName("/home/claude/apache/creadur-rat/apache-rat-core/target/test-classes/GitIgnoreBuilderTest/src/dir1/file1.log")
+                .setBaseName("home/claude/apache/creadur-rat/apache-rat-core/target/test-classes/GitIgnoreBuilderTest/src/").build();
+        System.out.println("Decomposition for " + candidate);
+
+        assertThat(matcher.toString()).isEqualTo("matcherSet(or('included .gitignore', 'included .gitignore'), or('excluded .gitignore', **/.gitignore, 'excluded .gitignore', 'excluded " + globalGitIgnore.substring(1) + "'))");
```

Comment:
`globalGitIgnore.substring(1)` does not give the right presentation on Windows. I don't have easy access to a Windows machine for experimenting - could someone help figure out the right incantation here?

Reply:
@raboof not sure if you see the test failure results, but it is:

```
Error:  Failures: 
Error:    GitIgnoreBuilderTest.test_global_gitignore:177 
expected: "matcherSet(or('included .gitignore', 'included .gitignore'), or('excluded .gitignore', **/.gitignore, 'excluded .gitignore', 'excluded D:/a/creadur-rat/creadur-rat/apache-rat-core/target/test-classes/GitIgnoreBuilderTest/global-gitignore'))"
 but was: "matcherSet(or('included .gitignore', 'included .gitignore'), or('excluded .gitignore', **/.gitignore, 'excluded .gitignore', 'excluded a/creadur-rat/creadur-rat/apache-rat-core/target/test-classes/GitIgnoreBuilderTest/global-gitignore'))"
```

Reply:
Thanks! That seems to have worked (though I did seem to need a `substring`)